guides/doc-Provisioning_Guide/topics/Virt-VMWare.adoc

```diff
@@ -406,6 +406,16 @@ You can now create an image from this virtual machine.
 
 You can use the xref:Provisioning_Virtual_Machines_in_VMware_vSphere-Adding_VMware_vSphere_Images_on_the_Satellite_Server[] section to add the image to {Project}.
 
+.Configuring {SmartProxy} to forward user data template
+
+{Project} must recognize the client IP address in order to serve correct template payload and when clients connects directly to {ProjectServer}, remote IP of the HTTP request can be used. However, when {Project} is deployed with {SmartProxy} templates feature, the callback URL is set to {SmartProxy} endpoint which forwards to the {ProjectServer} with HTTP header information X-Forwarded-For.
+
+For security reasons, {Project} only accepts this HTTP header from localhost. For each individual {SmartProxy}, a regular expression in *Administer* > *Settings* > *Provisioning* > *Remote address* must be updated to recognize IP address. From CLI this can be done using the following command:
+
+[options="nowrap" subs="+quotes,attributes"]
+----
+# hammer settings set --name remote_addr --value '^192\.0\.2\.(1|2|3)$'
```

Comment:
I didn't test this, but with Puma localhost is also a remote proxy which will send `HTTP_X_FORWARDED_FOR`. I think that needs to be always present, or you break non-capsule provisioning.

Reply:
Which brings us to IPv6 - this should probably be something like `localhost(4|6|4to6)?`. I don't exactly know when OS decides to prefer IPv6 over Ipv4 but on my PC when I connect to Rails (3000 TCP) it uses localhost4to6 address actually.

Amended.